app/persistence/schedule/schedule_test.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/entity/schedule"
+	"github.com/limechain/hedera-eth-bridge-validator/app/persistence/entity/status"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	"github.com/limechain/hedera-eth-bridge-validator/test/helper"
+	"github.com/limechain/hedera-eth-bridge-validator/test/mocks"
+	"github.com/stretchr/testify/assert"
+	"gorm.io/gorm"
+	"regexp"
+	"testing"
+)
+
+var (
+	repository   *Repository
+	dbConnection *gorm.DB
+	sqlMock      sqlmock.Sqlmock
+	db           *sql.DB
+
+	insertQuery                 = regexp.QuoteMeta(`INSERT INTO "schedules" ("transaction_id","schedule_id","has_receiver","operation","status","transfer_id") VALUES ($1,$2,$3,$4,$5,$6)`)
+	updateStatusQuery           = regexp.QuoteMeta(`UPDATE "schedules" SET "status"=$1 WHERE transaction_id = $2`)
+	selectQuery                 = regexp.QuoteMeta(`SELECT * FROM "schedules" WHERE transaction_id = $1 ORDER BY "schedules"."transaction_id" LIMIT 1`)
+	selectIdsByStatusQuery      = regexp.QuoteMeta(`SELECT "transaction_id" FROM "schedules" WHERE status = $1`)
+	selectReceiverTransferQuery = regexp.QuoteMeta(`SELECT * FROM "schedules" WHERE transfer_id = $1 AND operation = $2 AND has_receiver = true ORDER BY "schedules"."transaction_id" LIMIT 1`)
+
+	transactionId  = "someTransactionId"
+	scheduleId     = "someScheduleId"
+	hasReceiver    = true
+	operation      = "someOperation"
+	expectedStatus = status.Submitted
+	transferId     = sql.NullString{String: "someTransferId", Valid: true}
+
+	entityColumns = []string{"transaction_id", "schedule_id", "has_receiver", "operation", "status", "transfer_id"}
+	entityArgs    = []driver.Value{transactionId, scheduleId, hasReceiver, operation, expectedStatus, transferId}
+
+	expectedSchedule = &entity.Schedule{
+		TransactionID: transactionId,
+		ScheduleID:    scheduleId,
+		HasReceiver:   hasReceiver,
+		Operation:     operation,
+		Status:        expectedStatus,
+		TransferID:    transferId,
+	}
+)
+
+func Test_NewRepository(t *testing.T) {
+	setup()
+
+	actualRepository := NewRepository(dbConnection)
+
+	assert.Equal(t, repository, actualRepository)
+}
+
+func Test_Create(t *testing.T) {
+	setup()
+	defer helper.CheckSqlMockExpectationsMet(sqlMock, t)
+	helper.SqlMockPrepareExec(sqlMock, insertQuery, transactionId, scheduleId, hasReceiver, operation, expectedStatus, transferId)
+
+	err := repository.Create(expectedSchedule)
+
+	assert.Nil(t, err)
+}
+
+func Test_Create_Error(t *testing.T) {
+	setup()
+	defer helper.CheckSqlMockExpectationsMet(sqlMock, t)
+	expectedErr := helper.SqlMockPrepareExecWithErr(sqlMock, insertQuery, transactionId, scheduleId, hasReceiver, operation, expectedStatus, transferId)
+
+	err := repository.Create(expectedSchedule)
+
+	assert.Error(t, err, expectedErr)
+}
+
+func Test_UpdateStatusCompleted(t *testing.T) {
+	setup()
+	defer helper.CheckSqlMockExpectationsMet(sqlMock, t)
+	updatedStatus := status.Completed
```

Comment:
Do we need the `updatedStatus` as a standalone variable in the function? Maybe we can use it directly as in `Test_UpdateStatusCompleted_Error` or to declare it at the beginning as the `expectedStatus` and re-used it in both `Test_UpdateStatusCompleted` and `Test_UpdateStatusCompleted_Error`?